waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -97,11 +97,6 @@ private void add(AbstractMetaStore metaStore) {
 
     if (metaStore.getFederationType() == PRIMARY) {
       primaryDatabaseMapping = databaseMapping;
-      if (!metaStoreMapping.isAvailable()) {
```

Comment:
This was added as part of Kerberos support: https://github.com/ExpediaGroup/waggle-dance/pull/266/files#diff-8c2a82b373b4f06981a5086c6e6f2cbafea8480451366fcf1ba3c1490450388a

For non Kerberos this is not needed and unwanted as Healtcheck connections just opening the processor will cause this to call the primary metastore. In production environments this is slowing things down. I moved it to the SASL code path.